### ISSUE TITLE
Facebook API - cannot access to mail

### ISSUE THREAD

**Issue by xunlao** [authomatic/authomatic#130]
I can only got name and ID from the facebook API call. In facebook's "LOGIN PERMISSIONS", I do see "email"; and in the config I set

``` python
'fb': {

        'class_': oauth2.Facebook,

        # Facebook is an AuthorizationProvider too.
        'consumer_key': 'MYSTUFF',
        'consumer_secret': 'MYSTUFF',

        # But it is also an OAuth 2.0 provider and it needs scope.
        'scope': ['email'],
    },
```

I got `result.user.data` to be  `{'name': 'Sean Lao', 'id': '55047288*******'}` -- which is good, but -- no email!? - and `result.user.email` is `None`, while `result.user.id, result.user.name` work.

The result is obtained through here:

``` python
result = authomatic.login(WerkzeugAdapter(request, response), provider_name)
```

How do I get the email?

P.s. it works only for Yahoo so far. Haven't tried on Twitter or Google yet.

and btw this is `result.user.to_dict()`:

``` python
{  
   'nickname':None,
   'provider':'fb',
   'first_name':None,
   'birth_date':'None',
   'id':'MYSTUFF',
   'location':None,
   'country':None,
   'credentials':'2%0A2-5%MYSTUFF',
   'phone':None,
   'data':{  
      'id':'MYSTUFF',
      'name':'Sean Lao'
   },
   'name':'Sean Lao',
   'locale':None,
   'city':None,
   'timezone':None,
   'last_name':None,
   'postal_code':None,
   'email':None,
   'username':None,
   'picture':'http://graph.facebook.com/THIS_ONE_WORKS/picture?type=large',
   'gender':None,
   'link':None,
   'gae_user':None
}
```

the `'picture':'http://graph.facebook.com/THIS_ONE_WORKS/picture?type=large',`
is working, though.

**Comment by xunlao** [comment]
anyone else having this problem?

**Comment by OpaVote** [comment]
This is what my setup looks like and I am able to get email addresses.

    facebook = dict(
        class_=oauth2.Facebook,
        consumer_key="XXX",
        consumer_secret="XXX",
        id=authomatic.provider_id(),
        scope=["user_about_me", "email"]
        ),

**Comment by xunlao** [comment]
wired.. this is my new setting but it just don't return email as usual.
```python
CONFIG = {
    'fb': {
        'id': 2,
        'class_': oauth2.Facebook,

        # Facebook is an AuthorizationProvider too.
        'consumer_key': 'X',
        'consumer_secret': 'X',

        'scope': ["user_about_me", "email", "public_profile"]
    },

    'oi': {'class_': openid.OpenID, 'id': 1, }
}
```

**Comment by emreozen** [comment]
did you figure this out? I can get everything except the e-mail. similar problem 

Thanks in advance

**Comment by xunlao** [comment]
I did not figure it out. I am considering using JavaScript API and use RESTful to send the FB login info back to Python.

**Comment by OpaVote** [comment]
I downloaded authomatic on Aug. 23.  I suppose it is possible that something has changed since then that broke getting FB emails.  You may want to try the code from Aug. 23.

**Comment by goschtl** [comment]
I guess the problem is the api call to facebooks graph api.
This is the url from authomatic https://graph.facebook.com/v2.3/me
If you change it to this.
https://graph.facebook.com/v2.3/me?fields=email
It will return the Email. 
What will be the best way to make this configurable.?

**Comment by fairbairn** [comment]
It actually has to do with the API version your specific Facebook App is bound to.  Once you get into 2.3, 2.4, and now 2.5, you have to explictly ask for the additional fields with the ?fields= parameter.  This is why some people are working and some are not.  It's about which Graph API both the App and Authomatic are supporting at the same time.

**Comment by thar** [comment]
Not sure if this comes too late, but you can follow the next workaround.
Simply put the next sentences before your config:

`# Facebook api workaround`
`oauth2.Facebook.user_info_url = 'https://graph.facebook.com/v2.5/me?fields=id,first_name,last_name,picture,email,gender,timezone,location,middle_name,name_format,third_party_id,website,birthday,locale'`

You can tune the fields you are asking to the graph api. The ones I've written are more than the ones authomatic handle.
You can also tune the graph api version.

**Comment by mrichar1** [comment]
It looks like these issues are being dealt with in PR #164 - testing and feedback on that PR welcome!

**Comment by tomster** [comment]
FTR #164 doesn't seem to address this. it only affects the default scope. even when setting the correct scope explicitly, i only get the name, not the email.

**Comment by mrichar1** [comment]
Apologies for misreading the scope of that PR. I've added a simple PR to explicitly request the fields that were previously provided implicitly (having first merged #164). Comments and testing welcome!
